compiler/rustc_ast/src/ast.rs

```diff
@@ -2191,6 +2191,14 @@ pub enum TraitObjectSyntax {
     None,
 }
 
+#[derive(Clone, PartialEq, Encodable, Decodable, Debug)]
+pub enum PreciseCapturingArg {
+    /// Lifetime parameter
+    Lifetime(Lifetime),
+    /// Type or const parameter
+    Arg(Ident, NodeId),
```

Comment:
Actually, I wonder if all of this is overkill and we can just parse this as a `Path` (an invariant that it's one segment, no args).

Reply:
Path ended up cleaning things up in the resolver!